### PR TITLE
Roles & Permissions performance improvements - 1

### PR DESCRIPTION
**Pull request by prakash-alpine**
This is for code review purpose only. Do not merge with Master.

### CODE REVIEW

File: app/mixins/permissioner.rb [hunk 1]
```diff
@@ -4,13 +4,41 @@
...
+    begin
+      if self.class.name != 'Events::Base' && self.chorus_scope == nil
+        Chorus.log_error("No chorus scope found for #{self.class.name}")
+      elsif self.class.name != 'Events::Base'
+        @attributes['chorus_scope_id'] = self.chorus_scope.id
+      else
+        Chorus.log_error("---- chorus scope does not exists for #{self.class.name} ----")
+        @attributes['chorus_scope_id'] = ChorusScope.default_chorus_scope.id
+      end
+    rescue Exception => e
```
Comment (AndrewLngdn) [inline app/mixins/permissioner.rb]: What kind of exceptions were you running into here?

**Comment by AndrewLngdn** [comment]
Looks good to me

File: app/mixins/permissioner.rb (hunk 1 above)
Comment (prakash-alpine) [inline app/mixins/permissioner.rb]: The Events:Base class does not have a real object in DB and hence it does not have the 'id' attribute which throws an exception.  I think this is the only place where I had to make an exception.

**Comment by prakash-alpine** [comment]
Do not merge this with master yet. Deploy directly from this branch onto 10.0.0.165

**Pull request closed without merging**